plotting_helpers.py

```diff
@@ -0,0 +1,728 @@
+
+
+@dataclass
+class LoadedResults:
+    """
+    A class to store loaded results.
+
+    Attributes
+    ----------
+    data_df : pd.DataFrame
+        A DataFrame containing the data.
+    collated_notes_df : pd.DataFrame
+        A DataFrame containing notes about
+        the run that generated these results.
+    """
+
+    data_df: pd.DataFrame()
+    collated_notes_df: pd.DataFrame()
+
+
+@dataclass
+class RealizationName:
+    """
+    Class to store the names of the seismicity rate model and ground motion
+    characterization model used in a realization.
+    """
+
+    seismicity_rate_model_id: str
+    ground_motion_characterization_models_id: str
+
+
+################################################
+### functions
+
+
+def convert_edge_margin_in_pixels_to_fraction(
+    fig: matplotlib.figure.Figure,
+    left_margin_pixels: int,
+    right_margin_pixels: int,
+    bottom_margin_pixels: int,
+    top_margin_pixels: int,
+) -> MarginFractions:
+    """
+    Convert edge margins from pixels to fractions of the figure dimensions.
+
+    Parameters
+    ----------
+    fig : matplotlib.figure.Figure
+        The figure object to get dimensions from.
+    left_margin_pixels : int
+        The left margin in pixels.
+    right_margin_pixels : int
+        The right margin in pixels.
+    bottom_margin_pixels : int
+        The bottom margin in pixels.
+    top_margin_pixels : int
+        The top margin in pixels.
+
+    Returns
+    -------
+    MarginFractions
+        A dataclass containing the margins as fractions of the figure dimensions.
+    """
+
+    # Get the figure height and width in inches and convert it to pixels
+    fig_width_inch = fig.get_figwidth()
+    fig_height_inch = fig.get_figheight()
+
+    dpi = fig.get_dpi()
+
+    fig_width_px = fig_width_inch * dpi
+    fig_height_px = fig_height_inch * dpi
+
+    left_margin_fraction = left_margin_pixels / fig_width_px
+    right_margin_fraction = 1.0 - right_margin_pixels / fig_width_px
+    bottom_margin_fraction = bottom_margin_pixels / fig_height_px
+    top_margin_fraction = 1.0 - top_margin_pixels / fig_height_px
+
+    return MarginFractions(
+        left_margin_fraction,
+        right_margin_fraction,
+        bottom_margin_fraction,
+        top_margin_fraction,
+    )
+
+
+def remove_special_characters(
+    s: str, chars_to_remove: tuple[str] = ("'", "[", "]", '"')
+) -> str:
+    """
+    Remove specified special characters from a string.
+
+    Parameters
+    ----------
+    s : str
+        The input string from which to remove characters.
+    chars_to_remove : tuple[str], optional
+        Characters to remove from the input string. Default is ("'", "[", "]", '"').
+
+    Returns
+    -------
+    str
+        The string with the specified characters removed.
+    """
+    translation_table = str.maketrans("", "", "".join(chars_to_remove))
+    return s.translate(translation_table)
+
+
+def load_aggregate_stats_for_one_logic_tree_one_location(
```

Comment:
Are these loading functions specifically for plotting?

Reply:
Moved the loading functions to their own module